server/src/routes/course/schedule.ts

```diff
@@ -38,3 +41,39 @@ export const getScheduleAsCsv = (_: ILogger) => async (ctx: Router.RouterContext
 
   setCsvResponse(ctx, OK, csv, `schedule_${courseId}`);
 };
+
+export const setScheduleFromCsv = (_: ILogger) => async (ctx: Router.RouterContext) => {
+  const courseId = Number(ctx.params.courseId);
+  const data = ctx.request.body;
+  
+  const isOldTasks = (entity: any) => entity.entityType === 'task' && entity.id;
+  const isNewTasks = (entity: any) => entity.entityType === 'task' && !entity.id;
+  const isOldEvents = (entity: any) => entity.entityType === 'event' && entity.id;
+  const isNewEvents = (entity: any) => entity.entityType === 'event' && !entity.id;
+
+  // we make 100500 requests in parallel and process them. But it's better to do it all in one transaction.
+  const result = await Promise.allSettled(data.map(async (entity: any) => {
```

Comment:
We can use `bulk update` approach. These may help: https://stackoverflow.com/questions/53790994/bulk-update-via-raw-query-in-typeorm https://github.com/typeorm/typeorm/issues/4934